### ISSUE TITLE
Add compatibility with Ralink MT7601U 148f:7601 Wi-Fi adapter

### ISSUE THREAD

**Issue by Elray02** [BelaPlatform/bela-image#14]
on the newest version of debian the driver for this device family are already installed, is it possible add it to the future version?

**Comment by giuliomoro** [comment]
Will look into it. Will you be available for testing, once done?

**Comment by Elray02** [comment]
yes sure, now i'm trying to use the wifi with the latest img for the bbb, with out the bela.

I have tryed to fix this thing alone but without succes.

Let me know if i can help 

UPDATE : 

with the last debian img for bbb the wifi work as plug and play.

**Comment by giuliomoro** [comment]
Alright try downloading this file, 
[mt7601usta.zip](https://github.com/BelaPlatform/bela-image/files/579752/mt7601usta.zip)
copy it to the device, unzip it, cd into `mt7601usta/src` and insert the module. That is, from a terminal, run the following"

``` bash
scp /path/to/mt7601usta.zip root@192.168.7.2:
ssh root@192.168.7.2
unzip mt7601usta.zip
insmod mt7601usta/src/os/linux/mt7601Usta.ko
```

if you don't get any error, then you should be able to do

``` bash
ifconfig -a
```

and see your wifi device (make sure you plugged in the wifi dongle before powering up the BeagleBone).

If the device shows up, then you can install the module so that it will automatically be loaded upon reboot:

``` bash
cd mt7601usta/src/
make install
```

To set your wifi settings, check out the corresponding page on [the wiki](https://github.com/BelaPlatform/Bela/wiki/Connecting-Bela-to-wifi) (or any generic Debian wifi tutorial).

**Comment by Elray02** [comment]
It work! great!
thanks!

**Comment by giuliomoro** [comment]
Hi @Elray02 I just added this driver to the [latest image](https://github.com/BelaPlatform/bela-image/releases/edit/v0.2.0b_stable). Could you please flash it and test it for me so I can close the issue?

**Comment by Elray02** [comment]
Yes of course :)

**Comment by Elray02** [comment]
it work perfectly thanks!

**Comment by giuliomoro** [comment]
Lovely. Closed v0.2.0b

**Comment by giuliomoro** [comment]
Lovely. Closed v0.2.0b

**Comment by Mridul1129261** [comment]
Hey @giuliomoro I am receiving this error

kex_exchange_identification: read: Connection reset by peer
lost connection

 Can you help me out ?

**Comment by giuliomoro** [comment]
no idea ... what image, what device, what driver ?

**Comment by Mridul1129261** [comment]
> no idea ... what image, what device, what driver ?

@giuliomoro I am using dual booted Kali Linux 2020.1 with Windows 10 as primary operating system in my Dell latitude e6410. Since Linux is a secondary operating system in my laptop, I am unable to get my preinstalled internal wifi adapters working in it. I have purchased an external wifi adapter whose chipset is mt7601u by Ralink. The command lsusb shows the adapter in the list but I can't see any wifi icon in the shortcut menu. I tried using your commands after downloading the zip file i.e. 

scp /path/to/mt7601usta.zip root@192.168.7.2:
ssh root@192.168.7.2
unzip mt7601usta.zip
insmod mt7601usta/src/os/linux/mt7601Usta.ko 

and received the following error

kex_exchange_identification: read: Connection reset by peer
lost connection

If you need to see output of any commands like dmesg or lspci on my terminal, please let me know.

**Comment by giuliomoro** [comment]
Right, so you are very much off topic here, not only because this repo is no longer maintained, but also because it is about a specific embedded platform for real-time audio and sensor processing called "Bela", which runs on the BeagleBone Black, which has an ARM CPU, and - in the version that this thread was referring to - was running Debian 7 with Linux 3.8.13. So, of the 4 lines you ran:

The first two
```
scp /path/to/mt7601usta.zip root@192.168.7.2:
ssh root@192.168.7.2
```
deal with copying the zip archive to the device (assuming it is available on the network at 192.168.7.2), which you won't need because you'd download it straight on your computer

This one
```
unzip mt7601usta.zip
```
unzips the archive, giving you a folder structure which contains the driver. Note, the driver is compiled for Linux 3.8.13, architecture armv7l.

The next line
```
insmod mt7601usta/src/os/linux/mt7601Usta.ko
```
would then try to load the module. Now, even assuming you got to this point, this line would fail because of the wrong architecture and wrong Linux version.

This driver I compiled from [this](https://github.com/art567/mt7601usta) repo probably. The readme on that repo now says 

> Note: Kernel 4.2 has been released which contains a driver for mt7601u, this repository is now deprecated.

Kali 2020.1 comes with Kernel 5.4.0, so - according to the line above - it should already contain the drivers for the device and there should be no need to build or install your own drivers. On Debian, I would do `ifconfig` or `ip a` to verify that the device is listed among the available network adapters. I never used Kali and I am not going to be able to provide any support on this. Ask the Kali people instead.
